packages/dd-trace/src/appsec/iast/iast-plugin.js

```diff
@@ -60,24 +60,10 @@ class IastPlugin extends Plugin {
     this.pluginSubs = []
   }
 
-  _wrapHandler (handler) {
-    return (message, name) => {
-      try {
-        handler(message, name)
-      } catch (e) {
-        log.error('[ASM] Error executing IAST plugin handler', e)
-      }
-    }
-  }
-
   _getTelemetryHandler (iastSub) {
     return () => {
-      try {
-        const iastContext = getIastContext(storage.getStore())
-        iastSub.increaseExecuted(iastContext)
-      } catch (e) {
-        log.error('[ASM] Error increasing handler executed metrics', e)
-      }
+      const iastContext = getIastContext(storage.getStore())
```

Comment:
If we don't catch here the error `IastPlugin` will be disabled by `Plugin`.
Because it doesn't differentiate between handlers, if a `Plugin`'s handler fails the whole `Plugin` is disabled.

I guess an error here is kind unlikely but i'd catch it without disabling the `IastPlugin`.

Reply:
Yes this is exactly what @CarlesDD said, IMO:

1. Overriding Plugin's addSub to not add the failing subscription instead of disabling the whole plugin
2. we could discuss with APM about modifying `Plugin` class behavior to make it less aggressive (since this might benefit other plugins too)

Reply:
Yes but just to clarify, what I meant is that `IastPlugin` subscribes twice: one for `_wrapHandler` and another one for `_getTelemetryHandler`.
If we decide that is OK to disable the IastPlugin on an error, I'd only do for errors in `_wrapHandler` and not in `_getTelemetryHandler`.
`_getTelemetryHandler` only increments a metric so is something _optional_ for IastPlugin (that could continue running despite of an error in telemetry).

Reply:
ok ugaitz convinced me! :P let's disable the `IastPlugin` in both cases

Reply:
@iunanua @CarlesDD @uurien Do you think it's good to disable it? If yes I will push a commit to fix linter